### PR TITLE
[1LP][RFR] Fixing failure related to quadicon deactivation test case

### PR DESCRIPTION
**Pull request by anurag03**
Purpose or Intent
================
Making changes in the test case related to quadicon deactivation to resolve failures because of the changes made in the data property of quadicon.

{{pytest: -v -k 'test_cloudprovider_noquads or test_template_noquads or test_datastore_noquads or test_host_noquads or test_infraprovider_noquads or test_vm_noquads'}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit dd16da47d91d01de0b26537ad7f67d14100793b6:

`cfme/infrastructure/datastore.py`:
- :bangbang: Line 31:1: **E303** *too many blank lines (3)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e8b560c1c48ff7e9be81ac1b7201bdda14ef167a:

`cfme/common/vm_views.py`:
- :bangbang: Line 57:1: **E303** *too many blank lines (3)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/common/vm_views.py [hunk 1]
```diff
@@ -77,33 +80,36 @@ class JSInstanceEntity(JSBaseEntity):
     @property
     def data(self):
         data_dict = super(JSInstanceEntity, self).data
-        if 'quadicon' in data_dict and data_dict['quadicon']:
-            quad_data = document_fromstring(data_dict['quadicon'])
-            data_dict['os'] = quad_data.xpath(self.QUADRANT.format(pos="a"))[0].get('src')
-            data_dict['vendor'] = quad_data.xpath(self.QUADRANT.format(pos="c"))[0].get('src')
-            data_dict['no_snapshot'] = quad_data.xpath(self.QUADRANT.format(pos="d"))[0].text
+        try:
```
Comment (quarckster) [inline cfme/common/vm_views.py]: **QUESTION**
I don't understand why `try` block is required here? Where `IndexError` can be raised?

Comment (anurag03) [reply cfme/common/vm_views.py]: @quarckster 
When there will be no 'os', 'vendor' and 'no_snapshot' index present in data_dict then IndexError will be raised.

File: cfme/common/vm_views.py [hunk 2]
```diff
@@ -78,10 +81,13 @@ class JSInstanceEntity(JSBaseEntity):
     def data(self):
         data_dict = super(JSInstanceEntity, self).data
         if 'quadicon' in data_dict and data_dict['quadicon']:
```
Comment (quarckster) [inline cfme/common/vm_views.py]: **REQUIRED**
Just add one more condition and remove try-except block.
```python
if data_dict and 'quadicon' in data_dict and data_dict['quadicon']:
```

File: cfme/infrastructure/datastore.py [hunk 3]
```diff
@@ -75,11 +80,26 @@ class NonJSDatastoreEntity(NonJSBaseEntity):
     tile_entity = DatastoreTileIconEntity
 
 
+class JSDatastoreEntity(JSBaseEntity):
+    @property
+    def data(self):
+        data_dict = super(JSDatastoreEntity, self).data
+        try:
```
Comment (quarckster) [inline cfme/infrastructure/datastore.py]: **REQUIRED**
Here as well:
```python
if data_dict and 'quadicon' in data_dict and data_dict['quadicon']:
```

File: cfme/tests/configure/test_visual_cloud.py [hunk 4]
```diff
@@ -160,5 +160,5 @@ def test_cloud_start_page(request, appliance, start_page):
 def test_cloudprovider_noquads(request, set_cloud_provider_quad):
     view = navigate_to(CloudProvider, 'All')
     view.toolbar.view_selector.select("Grid View")
-    # Here get_first_entity() method will return None when the Quadrants option is deactivated.
-    assert view.entities.get_first_entity().data is None
+    # Here data property will return an empty dict when the Quadrants option is deactivated.
+    assert bool(view.entities.get_first_entity().data) is False
```
Comment (quarckster) [inline cfme/tests/configure/test_visual_cloud.py]: **REQURIED**
Here and below
```python
assert not view.entities.get_first_entity().data
```

**Comment by anurag03** [comment]
@quarckster : Indexerror occurs in 5.9, please check the trace back for 5.9. :)

**Comment by quarckster** [comment]
@anurag03 The traceback shows that issue in `quad_data.xpath(self.QUADRANT.format(pos="a"))[0].text`, please debug it.

**Comment by anurag03** [comment]
@quarckster : Implemented changes based on the changes made in PR: https://github.com/ManageIQ/integration_tests/pull/5871

File: cfme/common/vm_views.py (hunk 2 above)
Comment (RonnyPfannschmidt) [reply cfme/common/vm_views.py]: as far as i can tell it should be condensed down to `data_dict.get('quadicon')` - the `IndexError looks like it is related to the ux access - and cant be fixed with the extra ckeck

Comment (anurag03) [reply cfme/common/vm_views.py]: @RonnyPfannschmidt I have removed those changes, latest commit doesn't contain the extra check instead I have included a try and except block to check IndexError and return empty dict.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 33776c6c88ec006bb8f4cbfad7ebbce917ddf07b:

`cfme/tests/infrastructure/test_delete_infra_object.py`:
- :red_circle: Line 6:1: **F401** *'cfme.infrastructure.resource_pool' imported but unused*

`cfme/tests/intelligence/test_rss.py`:
- :red_circle: Line 5:1: **F401** *'cfme.fixtures.pytest_selenium as sel' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 10f7f3356dc11af33857a8c86c06025519b38a58:

`cfme/tests/infrastructure/test_delete_infra_object.py`:
- :red_circle: Line 6:1: **F401** *'cfme.infrastructure.resource_pool' imported but unused*

`cfme/tests/intelligence/test_rss.py`:
- :red_circle: Line 5:1: **F401** *'cfme.fixtures.pytest_selenium as sel' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/common/vm_views.py [hunk 5]
```diff
@@ -34,6 +48,8 @@ def data(self):
 
             state = os.path.split(state)[1]
             state = os.path.splitext(state)[0]
+        except NoSuchElementException:
```
Comment (izapolsk) [inline cfme/common/vm_views.py]: this old code looks weird but I don't think you need to fix this in this PR

Comment (anurag03) [reply cfme/common/vm_views.py]: Let me know the changes I will raise a PR. @izapolsk

**Pull request merged**